spring-integration-file/src/main/java/org/springframework/integration/file/remote/session/SessionFactoryMapBuilder.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.remote.session;
+
+import org.springframework.integration.support.MapBuilder;
+
+/**
+ * @author Gary Russell
+ * @since 5.0.7
+ *
+ */
+public class SessionFactoryMapBuilder<T> extends MapBuilder<SessionFactoryMapBuilder<T>, Object, SessionFactory<T>> {
```

Comment:
We need to think about retiring this builder in favor of `Map.of()`: https://stackoverflow.com/questions/46601959/what-is-the-use-of-map-ofentries-instead-of-map-of.

But, yeah... we still can live with that for a while! 😄 

Please, add some simple JavaDoc on the matter though

Reply:
`Map.of()` is JDK 9 - I don't think we'll be there for a long time.